### PR TITLE
feat(telemetry): add server side telemetry

### PR DESCRIPTION
**Pull request by RyanHolstien**
## Checklist
- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [X] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [X] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/entity/cassandra/CassandraEntityService.java [hunk 1]
```diff
@@ -331,6 +333,33 @@ protected List<Pair<String, UpdateAspectResult>> ingestAspectsToLocalDB(
     }, DEFAULT_MAX_TRANSACTION_RETRY);
   }
 
+  @Nullable
+  @Override
+  public RecordTemplate ingestAspectIfNotPresent(@Nonnull Urn urn, @Nonnull String aspectName,
+      @Nonnull RecordTemplate newValue, @Nonnull AuditStamp auditStamp, SystemMetadata systemMetadata) {
```
Comment (dexter-mh-lee) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/cassandra/CassandraEntityService.java:339]: nit: let's add annotation to system metadata for consistency

Comment (dexter-mh-lee) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/cassandra/CassandraEntityService.java:339]: @Nonnull ? I think?

Comment (RyanHolstien) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/cassandra/CassandraEntityService.java:339]: Yeah I think so, I'll add it onto ingestAspect too then, I copied the method definition from that 😆

File: metadata-models/src/main/pegasus/com/linkedin/common/ClientId.pdl [hunk 2]
```diff
@@ -0,0 +1,14 @@
+namespace com.linkedin.common
+
+/**
+* A simple wrapper around a String to persist the client ID for telemetry in DataHub's backend DB
+*/
+@Aspect = {
+    "name": "clientId"
+}
+record ClientId {
```
Comment (dexter-mh-lee) [inline metadata-models/src/main/pegasus/com/linkedin/common/ClientId.pdl:9]: Sorry about the renaming comment, but if we are going to use this specific for Telemetry, then I think we should name this TelemetryClientId and create a telemetry package and put it there. We been isolating aspects that bound to specific entities in it's own package (like policy / assertions / etc. )

File: metadata-models/src/main/pegasus/com/linkedin/metadata/key/TelemetryKey.pdl [hunk 3]
```diff
@@ -0,0 +1,14 @@
+namespace com.linkedin.metadata.key
+
+/**
+ * Key for a Tag
```
Comment (dexter-mh-lee) [inline metadata-models/src/main/pegasus/com/linkedin/metadata/key/TelemetryKey.pdl:4]: Can we add documentation saying that there should only be one key?

File: metadata-models/src/main/pegasus/com/linkedin/common/ClientId.pdl (hunk 2 above)
Comment (RyanHolstien) [reply metadata-models/src/main/pegasus/com/linkedin/common/ClientId.pdl:9]: No worries, can do 👍

File: metadata-models/src/main/pegasus/com/linkedin/metadata/key/TelemetryKey.pdl (hunk 3 above)
Comment (RyanHolstien) [reply metadata-models/src/main/pegasus/com/linkedin/metadata/key/TelemetryKey.pdl:4]: Yep!

Comment (RyanHolstien) [reply metadata-models/src/main/pegasus/com/linkedin/metadata/key/TelemetryKey.pdl:4]: This has a copy & paste error anyway, thanks for pointing it out 😄

**Review by dexter-mh-lee — APPROVED** [review]
LGTM!

**Pull request merged**